libraries/Microsoft.Bot.Builder/HealthCheck.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder
+{
+    internal class HealthCheck
+    {
+        public static HealthCheckResponse CreateHealthCheckResponse(IConnectorClient connector)
+        {
+            // A derived class may override this, however, the default is that the bot is healthy given we have got to here.
+            var healthResults = new HealthResults { Success = true };
+
+            if (connector != null)
+            {
+                try
+                {
+                    // This is a mock secure SendToConversation to grab the exact HTTP headers.
+                    // If you have no appId and no secret this code will run but no produce an Authorization header.
+                    var captureHandler = new CaptureRequestHandler();
+                    var client = new ConnectorClient(connector.BaseUri, connector.Credentials, captureHandler);
+                    var activity = new Activity { Type = "message", Conversation = new ConversationAccount { Id = "capture" } };
```

Comment:
We have a constant for this somewhere, might require an extra using

```suggestion
                    var activity = new Activity { Type = ActivityTypes.Message, Conversation = new ConversationAccount { Id = "capture" } };
```

Reply:
OK I should have used this constant. I'll fix tomorrow.